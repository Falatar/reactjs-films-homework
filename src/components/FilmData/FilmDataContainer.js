import { connect } from 'react-redux';
import showTrailer, { updatePath } from './actions';
import FilmData from './FilmData';


const mapDispatchToProps = (dispatch) => ({
  openModal: () => dispatch(showTrailer()),
  setTrailer: (path) => dispatch(updatePath(path)),
});

export default connect(
  null,
  mapDispatchToProps,
)(FilmData);
