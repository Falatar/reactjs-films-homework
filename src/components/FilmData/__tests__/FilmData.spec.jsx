import React from 'react';
import renderer from 'react-test-renderer';
import FilmDataContainer from '../FilmDataContainer';

it('renders correctly', () => {
  const tree = renderer
    .create(<FilmDataContainer imgURL="" id={1} name="nemo" tagList="empty" rating={0.0} overview="some text" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
