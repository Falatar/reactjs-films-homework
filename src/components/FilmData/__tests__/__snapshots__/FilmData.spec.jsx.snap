// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmData renders correctly 1`] = `
<div
  className="film__data"
>
  <div
    className="control"
  >
    <img
      alt={1}
      className="wall"
      onError={[Function]}
      onLoad={[Function]}
      src="https://image.tmdb.org/t/p/w500"
    />
    <div
      className="hidden"
    >
      <button
        className="play"
        onClick={[Function]}
        type="button"
      >
        â–¶
      </button>
      <p
        className="watch__now"
      >
        Watch Now
      </p>
      <button
        className="info"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
  </div>
  <div
    className="description"
  >
    <div
      className="title"
    >
      <h3
        className="name"
      >
        nemo
      </h3>
      <h2
        className="tags"
      >
        empty
      </h2>
    </div>
    <h1
      className="rate"
    >
      0.0
    </h1>
  </div>
</div>
`;
